nightingaleproject/vrdr-dotnet
# Fhir ije race ethnicity bug

This is for ticket [NVSS-96](https://tracker.codev.mitre.org/browse/NVSS-96)

Initially, we thought this was a bug, but the test files were just formatted incorrectly (race and ethnicity were swapped). See ticket for the original test files. The updated test files are attached. I also made a small change to the logic for Hispanic Origin other. The previous logic made hispanic origin other and ije specific origins like Puerto Rican mutually exclusive.

**Testing**
To test run the following on the two test files
```
dotnet run 2ije round-trip-works.json > round-trip-works.ije
dotnet run ije2json round-trip-works.ije > round-trip-works-v2.json
```
The results should match the original json within the constraints of ije.

I also added two test cases in DeathRecord_Should.cs so we can verify our logic works in the future. To test run the following in VRDR.Tests
```
dotnet test
```
All tests should pass.

[testfiles.zip](https://github.com/nightingaleproject/vrdr-dotnet/files/6664413/testfiles.zip)
